docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,603 @@
+how the different lit commands work.
+
+## Debugging issues
+
+In this section, we briefly describe two common ways of debugging: print
+debugging and using LLDB.
+
+Depending on the code you're interested in, LLDB may be significantly more
+effective when using a debug build. Depending on what components you are
+working on, you could turn off optimizations for only a few things.
+Here are some example invocations:
+
+```sh
+# optimized Stdlib + debug Swiftc + optimized Clang/LLVM
+utils/build-script --release-debuginfo --debug-swift # other flags...
+
+# debug Stdlib + optimized Swiftc + optimized Clang/LLVM
+utils/build-script --release-debuginfo --debug-swift-stdlib # other flags...
+
+# optimized Stdlib + debug Swiftc (expect typechecker) + optimized Clang/LLVM
+utils/build-script --release-debuginfo --debug-swift --force-optimized-typechecker
+
+# Last resort option, it is highly unlikely that you will need this
+# debug Stdlib + debug Swiftc + debug Clang/LLVM
+utils/build-script --debug # other flags...
+```
+
+Debug builds have two major drawbacks:
+- A debug compiler is much slower, leading to longer feedback loops in case you
+  need to repeatedly compile the Swift standard library and/or run a large
+  number of tests.
+- The build artifacts consume a lot more disk space.
+
+[DebuggingTheCompiler.md](/docs/DebuggingTheCompiler.md) goes into a LOT
+more detail on how you can level up your debugging skills! Make sure you check
+it out in case you're trying to debug a tricky issue and aren't sure how to
+go about it.
+
+### Print debugging
+
+A large number of types have `dump(..)`/`print(..)` methods which can be used
+along with `llvm::errs()` or other LLVM streams. For example, if you have a
+variable `std::vector<CanType> canTypes` that you want to print, you could do:
+```cpp
+auto &e = llvm::errs();
+e << "canTypes = [";
+llvm::interleave(canTypes,
+                 [&](auto ty) { ty.dump(e); },
+                 [&]() { e << ", "; });
+e << "]\n";
+```
+You can also crash the compiler using `assert`/`llvm_unreachable`/
+`llvm::report_fatal_error`, after accumulating the result in a stream:
+```cpp
+std::string msg; llvm::raw_string_ostream os(msg);
+os << "unexpected canTypes = [";
+llvm::interleave(canTypes,
+                 [&](auto ty) { ty.dump(os); },
+                 [&]() { os << ", "; });
+os << "] !!!\n";
+llvm::report_fatal_error(os.str());
+```
+
+### Debugging using LLDB
+
+When the compiler crashes, the commandline arguments passed to it will be
+printed to stderr. It will likely look something like:
+
+```
+/path/to/swift-frontend <args>
+```
+
+- Using LLDB on the commandline: Copy the entire invocation and pass it to LLDB.
+  ```sh
+  lldb -- /path/to/swift-frontend <args>
+  ```
+  Now you can use the usual LLDB commands like `run`, `breakpoint set` and so
+  on. If you are new to LLDB, check out the [official LLDB documentation][] and
+  [nesono's LLDB cheat sheet][].
+- Using LLDB within Xcode:
```

Comment:
We should mention that using `expr llvm::errs().enable_colors(false)` can be used to stop printing color codes to the Xcode terminal, which does not support color codes at the moment.